mp/src/game/client/neo/ui/neo_scoreboard.cpp

```diff
@@ -1,255 +1,774 @@
 		}
-		m_pPlayerList->AddColumnToSection(sectionID, "name", "", 0, scheme()->GetProportionalScaledValueEx( GetScheme(), m_iNameWidth) - m_iAvatarWidth, hFallbackFont );
+		pPlayerList->AddColumnToSection(sectionID, "name", "", 0, m_iNameWidth - m_iAvatarWidth );
 	}
 
 	// set the section to have the team color
-	if (GameResources())
+	pPlayerList->SetSectionFgColor(sectionID, (teamNumber == TEAM_SPECTATOR) ? COLOR_NEO_WHITE : GameResources()->GetTeamColor(teamNumber));
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Used for sorting players
+//-----------------------------------------------------------------------------
+bool CNEOScoreBoard::StaticPlayerSortFunc(vgui::SectionedListPanel *list, int itemID1, int itemID2)
+{
+	KeyValues *it1 = list->GetItemData(itemID1);
+	KeyValues *it2 = list->GetItemData(itemID2);
+	Assert(it1 && it2);
+
+	// first compare xp
+	int v1 = it1->GetInt("xp");
+	int v2 = it2->GetInt("xp");
+	if (v1 > v2)
+		return true;
+	else if (v1 < v2)
+		return false;
+
+	// next compare deaths
+	v1 = it1->GetInt("deaths");
+	v2 = it2->GetInt("deaths");
+	if (v1 > v2)
+		return false;
+	else if (v1 < v2)
+		return true;
+
+	// the same, so compare itemID's (as a sentinel value to get deterministic sorts)
+	return itemID1 < itemID2;
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Adds a new row to the scoreboard, from the playerinfo structure
+//-----------------------------------------------------------------------------
+bool CNEOScoreBoard::GetPlayerScoreInfo(int playerIndex, KeyValues *kv)
+{
+	kv->SetInt("playerIndex", playerIndex);
+	const int neoTeam = g_PR->GetTeam(playerIndex);
+	kv->SetInt("team", neoTeam);
+	kv->SetString("name", g_PR->GetPlayerName(playerIndex) );
+	kv->SetInt("deaths", g_PR->GetDeaths( playerIndex ));
+
+	const int xp = g_PR->GetXP(playerIndex);
+	const int neoClassIdx = g_PR->GetClass(playerIndex);
+	kv->SetString("rank", GetRankName(xp));
+	kv->SetInt("xp", xp);
+
+	CBasePlayer* player = C_BasePlayer::GetLocalPlayer();
+	const int playerNeoTeam = player->GetTeamNumber();
+	const bool oppositeTeam = (playerNeoTeam == TEAM_JINRAI || playerNeoTeam == TEAM_NSF) && (neoTeam != playerNeoTeam);
+
+	int statusIcon = -1;
+	if (neoTeam == TEAM_JINRAI || neoTeam == TEAM_NSF)
+	{
+		statusIcon = (!SHOW_ENEMY_STATUS && oppositeTeam || g_PR->IsAlive(playerIndex)) ? -1 : m_iDeadIcon;
+	}
+	kv->SetInt("status", statusIcon);
+	kv->SetString("class", oppositeTeam ? "" : GetNeoClassName(neoClassIdx));
+
+	if (g_PR->GetPing( playerIndex ) < 1)
+	{
+		if ( g_PR->IsFakePlayer( playerIndex ) )
+		{
+			kv->SetString("ping", "BOT");
+		}
+		else
+		{
+			kv->SetString("ping", "");
+		}
+	}
+	else
 	{
-		m_pPlayerList->SetSectionFgColor(sectionID, (teamNumber == TEAM_SPECTATOR) ? COLOR_NEO_WHITE : GameResources()->GetTeamColor(teamNumber));
+		kv->SetInt("ping", g_PR->GetPing( playerIndex ));
 	}
```

Comment:
I think this section could be made simpler if we just SetString("ping", "BOT") if IsFakePlayer, otherwise SetInt("ping", g_PR->GetPing(playerIndex));

Reply:
I'll change it but I wonder why the original code did that. In what cases other than bots can GetPing return a value < 1? 🤔